invenio_mail/tasks.py

```diff
@@ -79,4 +84,36 @@ def _send_email_with_attachments(data, attachments=[]):
             disposition = attachment.get("disposition")
         msg.attach(content_type=content_type, data=rawdata, disposition=disposition)
 
-    current_app.extensions["mail"].send(msg)
+    _send(msg, self)
+
+
+def _send(msg: Message, task):
```

Comment:
For the moment, we have decided not to use types hinting:
```suggestion
def _send(msg, task):
```